src/nunavut/lang/py/templates/base.j2

```diff
@@ -0,0 +1,424 @@
+
+    {%- elif f.data_type is FloatType %}
+        """Raises ValueError if the value is finite and outside of the permitted range, regardless of the cast mode."""
+        {#- We do not emit range check for float64 because its range matches that of the native Python's float. #}
+        {%- if f.data_type.bit_length < 64 %}
+        x = float(x)
+        in_range = {{ f.data_type.inclusive_value_range.min }}.0 <= x <= {{ f.data_type.inclusive_value_range.max }}.0
+        if in_range or not _np_.isfinite(x):
+            self._{{ f|id }} = x
+        else:
+            raise ValueError(f'{{ f|id }}: value {x} is not in [{{ f.data_type.inclusive_value_range.min }}, {# -#}
+                                                                {{ f.data_type.inclusive_value_range.max }}]')
+        {%- else %}
+        self._{{ f|id }} = float(x)  # Range check not required
+        {%- endif %}
+
+    {%- elif f.data_type is ArrayType %}
+        {{ assign_array(f, 'x') | indent(4) }}
+
+    {%- elif f.data_type is CompositeType %}
+        if isinstance(x, {{ f.data_type|full_reference_name }}):
+            self._{{ f|id }} = x
+        else:
+            raise ValueError(f'{{ f|id }}: expected {{ f.data_type|full_reference_name }} got {type(x).__name__}')
+
+    {%- else -%}{%- assert False -%}
+    {%- endif %}
+    {%- if type.inner_type is UnionType %}
+        {%- for z in type.fields if z.name != f.name %}
+        self._{{ z|id }} = None
+        {%- endfor %}
+    {%- endif %}
+{% endfor -%}
+{#
+ # SERIALIZATION METHODS
+ #}
+    # noinspection PyProtectedMember
+    def _serialize_(self, _ser_: _Serializer_) -> None:
+        {{ serialize(type) | remove_blank_lines | indent }}
+
+    # noinspection PyProtectedMember
+    @staticmethod
+    def _deserialize_(_des_: _Deserializer_) -> {{ full_class_name }}:
+        {{ deserialize(type, full_class_name) | remove_blank_lines | indent }}
+        assert isinstance(self, {{ full_class_name }})
+        return self
+{#
+ # PYTHON DATA MODEL
+ #}
+    def __repr__(self) -> str:
+{%- if type.inner_type is not UnionType %}
+        _o_0_ = ', '.join([
+    {%- for f in type.fields_except_padding %}
+            '{{ f.name }}=%s' % {{ printable_field_representation(f) }},
+    {%- endfor %}
+        ])
+{%- else %}  {#- UNION #}
+        _o_0_ = '(MALFORMED UNION)'
+    {%- for f in type.fields %}
+        if self.{{ f|id }} is not None:
+            _o_0_ = '{{ f.name }}=%s' % {{ printable_field_representation(f) }}
+    {%- endfor %}
+{%- endif %}
+        return f'{{ type.full_name }}.{{ type.version.major }}.{{ type.version.minor }}({_o_0_})'
+{#
+ # PYDSDL TYPE DESCRIPTOR
+ #}
+    {%- if T.has_fixed_port_id %}
+    _FIXED_PORT_ID_ = {{ T.fixed_port_id|int }}
+    {%- endif %}
+    {%- assert type.extent % 8 == 0 %}
+    _EXTENT_BYTES_ = {{ type.extent // 8 }}
+
+    {% set meta_type = type.__class__.__name__ -%}
+    _MODEL_: _pydsdl_.{{ meta_type }} = _restore_constant_(
```

Comment:
Let's bake in some comments here to explain why there's a big, scary, blob of opaque data in the generated python. It might concern some people from a security standpoint.